libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2780,31 +2792,39 @@ std::string YulUtilFunctions::readFromStorageDynamic(Type const& _type, bool _sp
 	});
 }
 
-std::string YulUtilFunctions::readFromStorageValueType(Type const& _type, std::optional<size_t> _offset, bool _splitFunctionTypes)
+std::string YulUtilFunctions::readFromStorageValueType(
+	Type const& _type,
+	std::optional<size_t> _offset,
+	bool _splitFunctionTypes,
+	VariableDeclaration::Location _location
+)
 {
 	solAssert(_type.isValueType(), "");
 
 	std::string functionName =
-			"read_from_storage_" +
-			std::string(_splitFunctionTypes ? "split_" : "") + (
-				_offset.has_value() ?
-				"offset_" + std::to_string(*_offset) :
-				"dynamic"
-			) +
-			"_" +
-			_type.identifier();
+		"read_from_" +
+		(_location == VariableDeclaration::Location::Transient ? "transient_"s : "") +
```

Comment:
> in IRGeneratorForStatements? There's already an assert at the end if it hits no other branches.

The problem is that it will hit other branches. The condition is `else if (m_context.isStateVariable(_variable))`, so any value other than `Transient` will be interpreted as storage, which is exactly the problem.

Reply:
You're right.